kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/RollableScalableResourceOperation.java

```diff
@@ -119,12 +119,12 @@ private void waitUntilScaled(final int count) {
     ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
     ScheduledFuture poller = executor.scheduleWithFixedDelay(tPoller, 0, POLL_INTERVAL_MS, TimeUnit.MILLISECONDS);
     try {
-      if (Utils.waitUntilReady(queue, rollingTimeout, rollingTimeUnit)) {
+      if (Utils.waitUntilReady(queue, getConfig().getScaleTimeout(), TimeUnit.MILLISECONDS)) {
```

Comment:
Why do you use getConfig().getScaleTimeout() and not read the this.scaleTimeout?

Regardless of the answer, this should be aligned with rollingTimeout. So we need to either use object properties for both or for none and read everything from the config.

Reply:
I more or less followed what was used [here](https://github.com/fabric8io/kubernetes-client/blob/9049073360054a260ff81c38eefa06e9ec982a70/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/DeploymentOperationsImpl.java#L219). The problem with the `rollingTimeout` was that it seemed to be always 0 (not set). I'm not sure whether there is some way how it should be configured instead of using the scale timeout.